compiler/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -499,7 +500,11 @@ fn add_method_to_struct_namespace(
     let type_module = struct_type.id.local_module_id();
     let module = &mut current_def_map.modules[type_module.0];
     module.declare_function(name_ident.clone(), func_id).map_err(|(first_def, second_def)| {
-        DefCollectorErrorKind::Duplicate { typ: DuplicateType::Function, first_def, second_def }
+        DefCollectorErrorKind::Duplicate {
+            typ: DuplicateType::TraitImplementation,
```

Comment:
Why would adding a method to a struct namespace lead to a duplicate trait implementation?

Reply:
This is outdated. Not it returns TraitAssosiatedFunction